### PR TITLE
Pass basePath in publish model instead of baseUrl

### PR DESCRIPTION
**Pull request by mingwli**
1. break baseUrl docfx config into hostName and basePath
2. rename siteBasePath to basePath to be consistent with OP
3. pass basePath inside publish model instead of baseUrl to avoid repeated combine/split

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5352)

### CODE REVIEW

**Comment by yufeih** [comment]
I'd like to keep the input term consistent with the output term, for input we currently use `base_url`

**Comment by live1206** [comment]
> I'd like to keep the input term consistent with the output term, for input we currently use `base_url`

That means we need to split `base url` while using it in Docs.Build and migration tool?

**Comment by yufeih** [comment]
> > I'd like to keep the input term consistent with the output term, for input we currently use `base_url`
> 
> That means we need to split `base_url` while using it in Docs.Build and migration tool?

Or change input `base_url` to `base_path` and `host`

**Comment by live1206** [comment]
> > > I'd like to keep the input term consistent with the output term, for input we currently use `base_url`
> > 
> > 
> > That means we need to split `base_url` while using it in Docs.Build and migration tool?
> 
> Or change input `base_url` to `base_path` and `host`
`host` is not consumed further, I agree with `base_path`

File: docs/specs/legacy.yml [hunk 1]
```diff
@@ -326,7 +328,8 @@ inputs:
     monikerRange:
       'docs/v1/**': '< netcore-2.0'
       'docs/v2/**': 'netcore-2.0'
-    baseUrl: https://docs.com/docs
+    hostName: https://docs.com
```
Comment (yufeih) [inline docs/specs/legacy.yml]: Technically `hostName` does not contain `https://`
```suggestion
    hostName: docs.com
```

File: docs/specs/legacy.yml [hunk 2]
```diff
@@ -326,7 +328,8 @@ inputs:
...
+    basePath: docs
```
Comment (yufeih) [inline docs/specs/legacy.yml]: `basePath` in docs starts with `/` but does not ends with `/` unless it is `/`, can ensure that in `Config.OnDeserialized` method

```suggestion
    basePath: /docs
```

File: docs/specs/output.yml [hunk 3]
```diff
@@ -21,15 +21,17 @@ outputs:
       ]
     }
 ---
-# Publish manifest contains following properties: name, product, baseUrl
+# Publish manifest contains following properties: name, product, site_base_path
```
Comment (live1206) [inline docs/specs/output.yml]:
```suggestion
# Publish manifest contains following properties: name, product, base_path
```

**Pull request merged**